### DISCUSSION TITLE
Using API to pull order details

### DISCUSSION THREAD

**Discussion by mbasch2** [etsy/open-api#798] [Q&A]
I requested API access in order to pull order data to that we can automate some of our internal production processes but was denied. It is only for our use and we aren't sending any information to a 3rd parts etc, simply putting data into our internal database (filemaker) that processes the automation portion, but I have been denied.  I am not a developer myself, so I am very confused why this would not be a valid use of the API.  The denial told me the request has to come from filemaker, but that is just a development environment with a database. It seems like this would be the most basic purpose of a customer API. What am I missing? Any advice on how to get API access?  Is there another way to pull order data?

**Comment by Makebuildmodify** [comment]
Hi, 

I don't know a lot about getting approved but my guess is the description that you wrote in the request was probably what lead them to the decision. I have a similar use case. I'm using the API to list my own shops listings and that's all. When I requested my first API keystring I wrote very little in my request. Something like, "This is for a script that I am writing for my own shop to list my own listing only.  It will not me used by anyone else." 

Anyway, you probably already have tried writing Etsy to ask why, but if you haven't yet, you could email the Etsy support team: developers@etsy.com

**Reply by mbasch2** [reply]
Thanks.  Yes, I am guessing it is something in the wording. I just can't figure out how to word it so they will consider it.  They seem to think I am passing information to app called "filemaker".  I don't thing they get that it is a development environment with a database.  I have tried reaching out but kept getting a canned response about my request violates their standards, but I can't see how pulling an order for internal processing does that. They finally just closed the request.  I was hoping to actually talk to a human being, but that doesn't seem to be the way Etsy operates.  Did you ever get your access?

**Comment by Makebuildmodify** [comment]
@mbasch2 

I'm still waiting on approval for my API v3 keystring. But I used a description very similar to my original API v2 request and added 2.0 to my original app name. That way they know I'm only updating my original app.  So, I'm pretty confident that I'll be approved.

If you're using it for you own shop and not sharing the data I would guess that they shouldn't really need to know what your process is after you interact with the API.

I would retry with a very simple description.  Something like, "pull order data to that we can automate some of our internal production processes".

**Reply by mbasch2** [reply]
thank you. I appreciate the help. I still can't figure out why they don't provided easy approval to access order info. It certainly seems that should be the basis of an API to allow stores to more efficiently interact.

**Reply by Makebuildmodify** [reply]
@mbasch2 
This is the description that I wrote for my first request: "This app is for listing many items at once. It creates listings in draft form. It's developed by me and for my use only. "

This is the description that I wrote for my second request for the new API: "This is the second iteration of my previous "Bulk Lister". I'm updating it to work with Oauth 2.0 and the Etsy API v3. I use it for my shop only. So, I'm both the User and the developer.  It's a tool to help me list in bulk. I have thousands unique of items and it's a huge help. "

I just got approval for the second request.
